dashboard/data/preprocess.py

```diff
@@ -17,44 +19,76 @@ def transform(self, X: np.ndarray) -> np.ndarray:
 class MergedAssaysPreprocessor:
     def __init__(
         self,
-        raw_compounds_df: pd.DataFrame,
-        chemical_columns: list[str],
-        key_column: str = "EOS",
     ):
         """
-        :param raw_compounds_df: compounds dataframe to be processed
-        :param chemical_columns: list of names of chemical colums
-        :param key_column: name of the key column, defaults to "EOS"
+        Parser for csv projection files.
         """
-        self.chemical_columns = chemical_columns
-        self.compounds_df = raw_compounds_df.set_index(key_column)
+        self.compounds_df = None
+        self.projection_df = None
+        self.columns_for_projection = None
+        self.chemical_columns = None  # TODO: incorporate chemical data
 
-    def restrict_to_chemicals(self) -> MergedAssaysPreprocessor:
+    def combine_assays_for_projections(
+        self, projection_files: tuple[str, io.StringIO], id_column: str = "EOS"
+    ) -> MergedAssaysPreprocessor:
+        processed_dfs = []
+        GROUP_BY_COLUMNS = [
+            "EOS",
+            "Source Plate Barcode",
+            "Source Well",
+            "Destination Plate Barcode",
+            "Destination Well",
```

Comment:
do we need source and destination? tried to use it with our initial data (from sem V) and there were just `Plate `and `Well` columns

Reply:
Yeah, though in the output file we have everything, hence the names.
Great that you pointed this out tho, I think it's wise to remove `GROUP_BY_COLUMNS` and group it using solely `EOS`, as it's an experiment from one screening.